src/v/kafka/server/handlers/find_coordinator.cc

```diff
@@ -22,149 +22,355 @@
+  const std::optional<ss::sstring>& error_message = std::nullopt) {
+    chunked_vector<coordinator> unauthorized_key_responses{};
+    unauthorized_key_responses.reserve(unauthorized_keys.size());
 
-    if (request.data.key_type == coordinator_type::transaction) {
-        if (!ctx.are_transactions_enabled()) {
-            return ctx.respond(
-              find_coordinator_response(error_code::unsupported_version));
-        }
+    std::ranges::transform(
+      std::move(unauthorized_keys),
+      std::back_inserter(unauthorized_key_responses),
+      [error_code, error_message](const KeyType& key) {
+          kafka::coordinator_response response_element{
+            key, error_code, error_message};
+          return response_element;
+      });
 
-        transactional_id tx_id(request.data.key);
+    return unauthorized_key_responses;
+}
 
-        auto authz = ctx.authorized(security::acl_operation::describe, tx_id);
+template<typename KeyType>
+using AuthorizedKeyHandler = std::function<ss::future<>(
+  request_context&,                   // request context
+  chunked_vector<KeyType>,            // authorized keys
+  chunked_vector<kafka::coordinator>& // out param
+  )>;
 
-        if (!ctx.audit()) {
-            return ctx.respond(find_coordinator_response(
-              error_code::broker_not_available,
-              "Broker not available - audit system failure"));
-        }
+// handler for the subset of authorized transaction ids
+static ss::future<> handle_authorized_txn_id(
+  request_context& ctx,
+  chunked_vector<transactional_id> authorized_keys,
+  chunked_vector<kafka::coordinator>& out_vector) {
+    co_await ss::max_concurrent_for_each(
+      std::move(authorized_keys),
+      max_concurrency,
+      [&ctx, &out_vector](auto authorized_key) {
+          return ctx.tx_gateway_frontend()
+            .find_coordinator(authorized_key)
+            .then([&ctx, &out_vector, authorized_key](const auto& response) {
+                kafka::coordinator_response response_element{
+                  authorized_key, error_code::coordinator_not_available};
+                auto maybe_leader = response.coordinator;
+                if (maybe_leader) {
+                    response_element = leader_to_coordinator(
+                      ctx, authorized_key, *maybe_leader);
+                }
+                out_vector.emplace_back(std::move(response_element));
+            });
+      });
+}
 
-        if (!authz) {
-            return ctx.respond(find_coordinator_response(
-              error_code::transactional_id_authorization_failed));
+// handler for the subset of authorized group ids
+static ss::future<> handle_authorized_group_id(
+  request_context& ctx,
+  chunked_vector<group_id> authorized_keys,
+  chunked_vector<kafka::coordinator>& out_vector) {
+    auto loop_body = [&ctx, &out_vector](group_id group_id) {
+        // if consumer group topic already exists, get the relevant partition
```

Comment:
there is an api to check if topic exists, maybe this can be simplified to something like this: 
```
ss::future<> f = ss::now();

if(!topic_exists){
    f= ctx.group_initializer().assure_topic_exists(true);
}

return f.then(....);
```
This way the logic to collect the out vector will only exists in a single `.then`